### ISSUE TITLE
problem for training data

### ISSUE THREAD

**Issue by slz929** [valeoai/MTAF#2]
Thanks for enlightening and practical work about multi-target DA !
I have read your paper, and I found one source dataset and 3 target datasets of unequal quantity, does the quantity of data for every domain matters? And what is the appropriate amount of training data for MTKT?
Another question, I want to know why KL loss is used for knowledge transfer? If I want to train an embedding word instead of a segmentation map, is the KL loss appropriate, and is there a better alternative?

**Comment by SportaXD** [comment]
Thank you for your interest in our work.
In our experiments, all the datasets have an unequal quantity of data. Overall, the quantity of data and the balance between the datasets really depend on the metrics you consider for your task. We propose for multi-target semantic segmentation to consider the mIoU individually over each target dataset and compute the average of those mIoUs in order for the metric to not have a bias towards the dataset with more test data. During training, we also make sure that we see as much data from each target dataset, even though we might need to do more epochs over the smaller datasets.
To sum up, it all depends on your task and metrics but you can find ways to overcome this inbalance between your datasets.
As for the KL loss, we need to have a way to compare and match two output distributions (the outputs of the target-specific and target-agnostic classifiers). For our task of semantic segmentation, we noticed that L2 loss also showed good results. You are free to use another loss depending on your task when it makes more sense.

**Comment by slz929** [comment]
thanks
